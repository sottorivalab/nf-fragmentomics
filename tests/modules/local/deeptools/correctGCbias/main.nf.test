nextflow_process {

    name "Test Process CORRECTGCBIAS"
    script "modules/local/deeptools/correctGCbias/main.nf"
    process "CORRECTGCBIAS"

    test("Should run CORRECTGCBIAS without failures") {
        setup {
            run("COMPUTEGCBIAS") {
                script "modules/local/deeptools/computeGCbias/main.nf"
                process {
                    """
                    input[0] = [
                        [ caseid:'test', sampleid: 'test', timepoint: 'T1' ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr22/sequence/chr22_23800000-23980000.2bit', checkIfExists: true)
                    ]
                    """
                }
            }
        }

        when {
            params {                
                outdir = "tests/results"
            }
            process {
                """
                input[0] = COMPUTEGCBIAS.out.freq
                """
            }
        }

        then {
            assert process.success
        }

    }

}
