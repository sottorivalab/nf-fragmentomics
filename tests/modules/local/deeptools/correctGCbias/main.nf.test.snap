{
    "Should run CORRECTGCBIAS without failures": {
        "content": [
            [
                [
                    {
                        "caseid": "test",
                        "sampleid": "test",
                        "timepoint": "T1"
                    },
                    "test.gc_correct.bam:md5,0d7537ab9988ee91ec451cf525246958",
                    "test.gc_correct.bam.bai:md5,5273cc612c92af0954129874f97022c3",
                    "test.freq.txt:md5,4a7eb3e068f25d4f021ca3740f30fe36"
                ]
            ],
            [
                "versions.yml:md5,954f95dfad3c0c454617079d7532ba7e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-22T10:59:09.024654"
    }
}